### DISCUSSION TITLE
Is there a way to have a 'before' function before a specific test?

### DISCUSSION THREAD

**Discussion by grantholle** [pestphp/pest#568] [Q&A]
I'm using datasets (which are amazing!), but I'm running into an issue where I'm trying to test some very specific edge cases on a piece of functionality. I would like the ability to have a `beforeX` function, just before one specific test.

Is there a way to do that on a `it()` function directly? The workaround is just to create an entire separate test file, but wondering if there was some sort of hook to do it for an individual test.

```php
it('can do something', function (data) {
    // Test
})->with([
    // My datasets
])->before(function () {
    // Some custom setup logic for this specific test
});
```

**Comment by dansysanalyst** [comment]
Hi @grantholle,

You could possibly have something like:


```php
<?php

beforeEach(function () {
    var_dump('Calls Before Each');

    if (str_contains($this->getName(), 'test #2')) {
        var_dump('runs only on test 2');
    }
});

test('test #1', function () {
    expect(true)->toBeTrue();
});

test('test #2', function ($str) {
    expect($str)->toBeString();
})->with(['a', 'b']);
```

Resulting in:

```plain
string(17) "Calls Before Each"

string(17) "Calls Before Each"

string(20) "runs only on test 2"

   PASS  Tests\Feature\ExampleTest
  ✓ test #1
  ✓ test #2 with ('a')
  ✓ test #2 with ('b')

  Tests:  3 passed
  Time:   1.81s
```

**Reply by grantholle** [reply]
Thanks. This could work.
